### PR TITLE
Implement tomaxint

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented `RandomState.tomaxint`.

### CODE REVIEW

File: cupy/random/generator.py [hunk 1]
```diff
@@ -223,6 +223,29 @@ def standard_normal(self, size=None, dtype=float):
...
+
+        """
+        if size is None:
+            size = ()
+        sample = cupy.empty(size, dtype='l')
+        # cupy.random only uses int32 random generator
+        curand.generate(
+            self._generator, sample.data.ptr, sample.size * 2)
+        # Disable sign bit
+        sample &= 0x7FFFFFFFFFFFFFFF
```
Comment (okuta) [inline cupy/random/generator.py]: Why do not use `sys.maxint`?
`sys.maxint` is different between Linux and Windows.

File: cupy/random/generator.py [hunk 2]
```diff
@@ -223,6 +223,29 @@ def standard_normal(self, size=None, dtype=float):
...
         return self.normal(size=size, dtype=dtype)
 
+    def tomaxint(self, size=None):
+        """Draws integers between 0 and max integer inclusive.
+
+        Args:
+            size (int or tuple of ints): Output shape.
+
+        Returns:
+            Drawn samples.
```
Comment (okuta) [inline cupy/random/generator.py]: Please add type

File: cupy/random/generator.py [hunk 3]
```diff
@@ -223,6 +223,29 @@ def standard_normal(self, size=None, dtype=float):
...
+        .. seealso::
+            :meth:`numpy.random.RandomState.tomaxint`
+
+        """
+        if size is None:
+            size = ()
+        sample = cupy.empty(size, dtype='l')
+        # cupy.random only uses int32 random generator
+        curand.generate(
+            self._generator, sample.data.ptr, sample.size * 2)
```
Comment (okuta) [inline cupy/random/generator.py]: We have to check size of sample to avoid index out of bounds.

**Comment by unnonouno** [comment]
I updated it.

**Comment by okuta** [comment]
I got error.
```
======================================================================
ERROR: test_tomaxint_int (test_generator.TestTomaxint)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/work/cupy/cupy/testing/random.py", line 116, in func
    f(self)
  File "/work/cupy/tests/cupy_tests/random_tests/test_generator.py", line 265, in setUp
    self.rs = generator.RandomState(seed=testing.generate_seed())
  File "/work/cupy/cupy/testing/random.py", line 81, in generate_seed
    return numpy.random.randint(0xffffffff, dtype=numpy.int64)
  File "mtrand.pyx", line 879, in mtrand.RandomState.randint (numpy/random/mtrand/mtrand.c:10256)
TypeError: randint() got an unexpected keyword argument 'dtype'
```

**Comment by unnonouno** [comment]
for this problem. https://github.com/cupy/cupy/pull/574

**Comment by unnonouno** [comment]
jenkins test this please

**Comment by okuta** [comment]
Cloud you rebase or merge msater?

**Comment by okuta** [comment]
LGTM!

**Pull request merged**